addons/hr_expense/models/hr_expense_sheet.py

```diff
@@ -657,6 +657,7 @@ def _do_reset_approval(self):
         self.activity_update()
 
     def _do_refuse(self, reason):
+        self.write({'approval_state': 'cancel'})
         self.write({'state': 'cancel'})
```

Comment:
@JulienAlardot I know why, the `compute_state` of expense sheet will trigger again when we change `approve_state` and of course the state of `expense` is computed base on `expense sheet` so i write like this, everything is ok and the failed test we 've seen earlier is pass, not sure for the other. If you ok with this approach i'll leave a code comment inside

Reply:
Please revert to the previous one, this won't fix anything. Calling write twice is bad practice, in theory we could just reset the approval state and the state would be recomputed but IMO we can already specify it here, it's less computation

Reply:
Ah i'm just being dump sr